vpr/SRC/route/route_timing.c

```diff
@@ -17,14 +18,31 @@
 #include "net_delay.h"
 #include "stats.h"
 #include "ReadOptions.h"
-
-
+ 
 using namespace std;
 
+/***************** Iterative connection based rerouting **********************/
 
+// conceptually works like rr_node_to_pin[net_index][sink_node_index] to get the pin index for that net
+// each net maps SINK node index -> PIN index for net
+// only need to be built once at the start since the SINK nodes never change
+// the reverse lookup of net_rr_terminals
+// do not query this directly, as rr_node_to_pin[inet][inode] will create the entry if it does not exist
+// instead, use it through functions convert_sink_node_to_pins_of_net and put_sink_rt_nodes_to_pins_lookup
+static vector<unordered_map<int,int>> rr_node_to_pin;
+
+// a property of each net, but only valid after pruning the previous route tree
+static vector<int> remaining_targets;
+static vector<t_rt_node*> reached_sinks;	// used to populate rt_node_of_sink after building route tree from traceback

```

Comment:
According to Oleg's suggestion, rr_sink_node_to_pin has been totally encapsulated in a Incremental_reroute_resources class. Its only internally relied upon for the public conversion and put in lookup operations (which the user supply a net index instead of a lookup table)